packages/apollo-server-core/src/utils/pluginTestHarness.ts

```diff
@@ -118,11 +119,17 @@ export default async function pluginTestHarness<TContext>({
     }
   }
 
+  const document = parse(graphqlRequest.query);
+  const operation = getOperationAST(document, graphqlRequest.operationName);
+  const operationName = operation?.name?.value;
   const requestContext: GraphQLRequestContext<TContext> = {
     logger: logger || console,
     schema,
     schemaHash: generateSchemaHash(schema),
     request: graphqlRequest,
+    document,
+    operation: operation ?? undefined,
+    operationName,
```

Comment:
@sachindshinde @jsegaran These changes need to be reverted or re-massaged because they violate the patterns which the plugin test harness is intending to to enforce, which is parity with that of the request pipeline.  It's possible that ships passing in the night led to this, but this is a principle that was meant to be had, and the placement that @glasser chose for assigning the `operation` in https://github.com/apollographql/apollo-server/pull/3465/files#diff-4c0ef897274fc1b61f7e31d4cbe29cc230b8f4e85e8ab4d6f8351feec33cb087R209-R212 is appropriate.

Put another way, the `operation` previously being assigned further down is very intentionally after validation and before we fire `didResolveOperation`.  Further, it is incorrect to assign the `operationName` to the request context without validating the document first (i.e., after we fire the callback returned from `validationDidStart`, which acts as a "`validationDidEnd`").  And even further, `document` is not to be set until after `didResolveSource`. 

These are all patterns which are baked into the [`requestPipeline.ts`](https://github.com/apollographql/apollo-server/blob/1bb4527a5b074489e1a2cabf40f89f6214bfefc7/packages/apollo-server-core/src/requestPipeline.ts#L284) and should be mirrored here precisely as it is those life-cycles that our plugin API ensures (by narrowing typing more precisely and guaranteeing the presence of these values at the appropriate stages of processing).

In a more durable world, these would be much more closely aligned with each other and it would be impossible to break that contract, but that is not the case today.

Reply:
To be clear, I am merely asking for these assignments of properties to the `requestContext` be done at the right phases which match that of the `requestPipeline.ts` file I linked, otherwise, we are testing things differently than they operate at runtime!

Reply:
@abernix Sounds good, thanks for the info around `pluginTestHarness.ts`. I've reverted this PR, I'll file a new PR that assigns `document`, `operation`, and `operationName` at the right phases (matching `requestPipeline.ts`) and add you as a reviewer.

Reply:
Thanks, @sachindshinde!  Let me know if I can help!